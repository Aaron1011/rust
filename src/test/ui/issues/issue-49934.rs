#![feature(stmt_expr_attributes)]
#![warn(unused_attributes)]

fn main() {
    // fold_stmt (Item)
    #[allow(dead_code)]
    #[derive(Debug)] // should not warn
    struct Foo;

    // fold_stmt (Mac)
    #[derive(Debug)]
    //~^ ERROR `derive` may only
    println!("Hello, world!");

    // fold_stmt (Semi)
    #[derive(Debug)] //~ ERROR `derive` may only
    "Hello, world!";

    // fold_stmt (Local)
    #[derive(Debug)] //~ ERROR `derive` may only
    let _ = "Hello, world!";

    // visit_expr
    let _ = #[derive(Debug)] "Hello, world!";
    //~^ ERROR `derive` may only

    let _ = [
        // filter_map_expr
        #[derive(Debug)] //~ ERROR `derive` may only
        "Hello, world!"
    ];
}
