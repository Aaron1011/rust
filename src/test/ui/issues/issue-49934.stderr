error[E0774]: `derive` may only be applied to structs, enums and unions
  --> $DIR/issue-49934.rs:11:5
   |
LL |     #[derive(Debug)]
   |     ^^^^^^^^^^^^^^^^

error[E0774]: `derive` may only be applied to structs, enums and unions
  --> $DIR/issue-49934.rs:16:5
   |
LL |     #[derive(Debug)]
   |     ^^^^^^^^^^^^^^^^

error[E0774]: `derive` may only be applied to structs, enums and unions
  --> $DIR/issue-49934.rs:20:5
   |
LL |     #[derive(Debug)]
   |     ^^^^^^^^^^^^^^^^

error[E0774]: `derive` may only be applied to structs, enums and unions
  --> $DIR/issue-49934.rs:24:13
   |
LL |     let _ = #[derive(Debug)] "Hello, world!";
   |             ^^^^^^^^^^^^^^^^

error[E0774]: `derive` may only be applied to structs, enums and unions
  --> $DIR/issue-49934.rs:29:9
   |
LL |         #[derive(Debug)]
   |         ^^^^^^^^^^^^^^^^

error: aborting due to 5 previous errors

For more information about this error, try `rustc --explain E0774`.
